go.mod

```diff
@@ -25,13 +25,15 @@ require (
 	github.com/gin-contrib/sse v0.0.0-20170109093832-22d885f9ecc7 // indirect
 	github.com/gin-gonic/gin v1.1.5-0.20170702092826-d459835d2b07 // indirect
 	github.com/go-sourcemap/sourcemap v2.1.2+incompatible // indirect
+	github.com/golang/protobuf v1.4.2
 	github.com/golang/snappy v0.0.0-20170215233205-553a64147049 // indirect
 	github.com/google/go-cmp v0.5.1 // indirect
 	github.com/gorilla/context v0.0.0-20160226214623-1ea25387ff6f // indirect
 	github.com/gorilla/mux v1.6.1 // indirect
 	github.com/gorilla/websocket v1.4.2
 	github.com/inconshreveable/mousetrap v1.0.0 // indirect
 	github.com/influxdata/influxdb1-client v0.0.0-20190402204710-8ff2fc3824fc
+	github.com/jhump/protoreflect v1.7.0
```

Comment:
I am a grpc and protobuf noob, but while reading up on them it seems like the newer google.golang.org/protobuf` does [actually support](https://blog.golang.org/protobuf-apiv2) what protoreflect is used for here? More specifically parsing protobuf definition during runtime and creating messages out of those definitions. Given this and the fact the one is the "standard" implementation, can we actually use that one? 

Maybe there are some additional problems that I am not aware of?

Reply:
This is true for the dynamic calls, but unfortunately it does not have the ability to parse the raw *.proto files. Google maintain that the C++ protoc tools is used for the parsing so there are few Go alternatives and protoreflect is the most complete choice.

Reply:
hm ... I guess nobody has done one that just parsers to the standard lib v2 interface types(if that is possible at all)? 

I guess this is the smaller of the big new dependencies that will be included, but still, I would prefer if would've liked it to use less ... if possible

Reply:
I have a parser that does just that; but it's not complete (other priorities) https://github.com/rogchap/protoparser

I opted for a production ready version

Reply:
@MStoykov @na-- 
So; to make sure we stay "up-to-date", I have refactored a lot of the code to use the Go Protobuf APIv2 and only use this library for the parsing only. This was everything is more standard and if we need to switch out the parser, that can be easily done.
As a side effect I've renamed the API from `Client.invokeRPC` to just `Client.invoke`; it's all RPC (duh) and this lines up with the grpc library under the hood (at the least the Go version)